src/picomsg/src/oper_db_writer.cpp

```diff
@@ -114,3 +201,306 @@ bool oper_db_writer_set_terminator_vrf_tunnel_counter(const uint32_t terminator_
+		PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to set oper DB object LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_ENCRYPTION_ALGO to %d",
+					oper_db_enc_algo);
+		return false;
+	}
+
+	enum LIBOPER_DB_CLIENT_FIELD_ENUM_VALUES oper_db_int_algo;
+	if(! int_algo_to_oper_db(oper_data->integrity_algorithm, oper_db_int_algo)) {
+		return false;
+	}
+
+	if (! db_obj->enum_set(db_obj.get(), LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_INTEGRITY_ALGO, oper_db_int_algo)) {
+		PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to set oper DB object LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_INTEGRITY_ALGO to %d",
+					oper_db_int_algo);
+		return false;
+	}
+
+	if (! db_obj->u64_set(db_obj.get(), LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_INBOUND_IPSEC_SPI, oper_data->inbound_ipsec_spi)) {
+		PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to set oper DB object LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_INBOUND_IPSEC_SPI to %lu",
+					oper_data->inbound_ipsec_spi);
+		return false;
+	}
+
+	if (! db_obj->u64_set(db_obj.get(), LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_OUTBOUND_IPSEC_SPI, oper_data->outbound_ipsec_spi)) {
+		PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to set oper DB object LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_OUTBOUND_IPSEC_SPI to %lu",
+					oper_data->outbound_ipsec_spi);
+		return false;
+	}
+
+	if (! db_obj->u64_set(db_obj.get(), LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_INBOUND_IKE_SPI, oper_data->inbound_ike_spi)) {
+		PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to set oper DB object LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_INBOUND_IKE_SPI to %lu",
+					oper_data->inbound_ike_spi);
+		return false;
+	}
+
+	if (! db_obj->u64_set(db_obj.get(), LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_OUTBOUND_IKE_SPI, oper_data->outbound_ike_spi)) {
+		PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to set oper DB object LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_OUTBOUND_IKE_SPI to %lu",
+					oper_data->outbound_ike_spi);
+		return false;
+	}
+
+	if (! db_obj->bool_set(db_obj.get(), LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_NAT_IDENTIFIED, oper_data->nat_identified)) {
+		PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to set oper DB object enum LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_NAT_IDENTIFIED to %d",
+					oper_data->nat_identified);
+		return false;
+	}
+
+	if (! db_obj->u64_set(db_obj.get(), LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_ANTI_REPLAY_WINDOW_SIZE, oper_data->anti_replay_window_size)) {
+		PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to set oper DB object LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_ANTI_REPLAY_WINDOW_SIZE to %lu",
+					oper_data->anti_replay_window_size);
+		return false;
+	}
+
+	if (! db_obj->bool_set(db_obj.get(), LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_EXT_SEQ_NUM, oper_data->ext_seq_num)) {
+		PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to set oper DB object enum LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_EXT_SEQ_NUM to %d",
+					oper_data->ext_seq_num);
+		return false;
+	}
+
+	struct liboper_union_str liboper_union_device_public_address;
```

Comment:
Also try to use designated initializer when initializing a struct. This makes the initialization more readable and in C++ actually gives a warning if you forgot to initialize one of the members
``` cpp
const struct liboper_union_str liboper_union_device_public_address = {
    .str = oper_data->device_public_address,
    .field_id = LIBOPER_DB_CLIENT_IPV4_ADDRESS,
};
```
Also the struct can be declared const :)
(ditto for the other variables of `struct liboper_union_str`)